src/plugins/home/server/services/sample_data/routes/uninstall.ts

```diff
@@ -78,9 +84,13 @@ export function createUninstallRoute(
         }
       }
 
-      const savedObjectsList = dataSourceId
-        ? sampleDataset.getDataSourceIntegratedSavedObjects(dataSourceId)
-        : sampleDataset.savedObjects;
+      let savedObjectsList = sampleDataset.savedObjects;
+      if (workspaceId) {
+        savedObjectsList = getWorkspaceIntegratedSavedObjects(savedObjectsList, workspaceId);
+      }
+      if (dataSourceId) {
+        savedObjectsList = getDataSourceIntegratedSavedObjects(savedObjectsList, dataSourceId);
+      }
```

Comment:
Should we take the order of these two `if`s into account?

Reply:
Yes, it will cause different id after condition switched.